### PR TITLE
Timebased bucketizer merge

### PR DESCRIPTION
**Pull request by ajuvercr**
Simplify writing sds:Records according to accessed buckets and newly created relations.

### CODE REVIEW

File: src/bucketizers/timebasedBucketizer.ts [hunk 1]
```diff
@@ -15,6 +14,8 @@ export default class TimebasedBucketizer implements Bucketizer {
     private readonly k: number = 4;
     private readonly minBucketSpan: number = 300000;
 
+    private readonly logger = getLogger("TimebasedBucketizer");
```
Comment (smessie) [inline src/bucketizers/timebasedBucketizer.ts:17]: That's the wrong logger. Would have been better if you would have started from #40

File: src/bucketizers.ts [hunk 2]
```diff
@@ -22,6 +22,44 @@ async function writeState(
     }
 }
 
+/* tslint:disable:no-unused-variable */
+async function prettyPrintQuads(quads: Quad[]): Promise<void> {
```
Comment (smessie) [inline src/bucketizers.ts:26]: Do we really want this here though?

File: src/bucketizers.ts [hunk 3]
```diff
@@ -278,62 +332,34 @@ export async function bucketize(
...
             const record_buckets = orchestrator.bucketize(
                 record,
                 buckets,
                 requestedBuckets,
                 newMembers,
+                newRelations,
                 sourceStream?.value || "root",
             );
+
+            record_buckets.forEach((x) => requestedBuckets.add(x));
```
Comment (smessie) [inline src/bucketizers.ts:353]: This record buckets should already be part of requested buckets by default, right? As you retrieve them in the bucketizers by using the getBucket function.

File: src/bucketizers/timebasedBucketizer.ts [hunk 4]
```diff
@@ -143,15 +146,17 @@ export default class TimebasedBucketizer implements Bucketizer {
                         parseInt(bucketProperties[1]) / this.k <
                         this.minBucketSpan
                     ) {
-                        console.log("We need to make a new page");
```
Comment (smessie) [inline src/bucketizers/timebasedBucketizer.ts:146]: I would keep these. Is helpful when debugging. But in #40 they are properly logged only in debug mode.

File: src/bucketizers.ts (hunk 3 above)
Comment (ajuvercr) [reply src/bucketizers.ts:353]: Not necessarily, because record_buckets are strings :cry:

**Pull request merged**